From 013963ba35e8fe8897211c0acf5ee98f9a871fc1 Mon Sep 17 00:00:00 2001
From: Michael Heimpold <mhei@heimpold.de>
Date: Fri, 10 Jan 2014 19:38:51 +0100
Subject: [PATCH] daemon_set_verbosity: fix erroneous error message

When calling this function with a value other than LOG_DEBUG
the error message was triggered erroneously.

Signed-off-by: Michael Heimpold <mhei@heimpold.de>
---

 libdaemon/dlog.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/libdaemon/dlog.c
+++ b/libdaemon/dlog.c
@@ -37,7 +37,7 @@ static int daemon_verbosity_level = LOG_
 void daemon_set_verbosity(int verbosity_prio) {
 
     /* Allow using negative verbosity levels to hide _all_ messages */
-    if (verbosity_prio > 0 && (verbosity_prio & LOG_PRIMASK) != LOG_PRIMASK)
+    if (verbosity_prio > 0 && (verbosity_prio & LOG_PRIMASK) != verbosity_prio)
         daemon_log(LOG_ERR, "The value %d is not a valid priority value", verbosity_prio);
 
     daemon_verbosity_level = verbosity_prio & LOG_PRIMASK;
